Binary/Locator/FileSystemLocator.php

```diff
@@ -23,12 +23,15 @@ class FileSystemLocator implements LocatorInterface
 
     /**
      * @param string[] $roots
+     * @param bool     $allowUnresolvable
      */
-    public function __construct(array $roots = [])
+    public function __construct(array $roots = [], bool $allowUnresolvable = false)
     {
-        $this->roots = array_map(function (string $root): string {
-            return $this->sanitizeRootPath($root);
-        }, $roots);
+        $this->roots = array_filter(array_map(function (string $root) use ($allowUnresolvable): ?string {
+            return $this->sanitizeRootPath($root, $allowUnresolvable);
+        }, $roots), function (string $root = null): bool {
+            return null !== $root;
+        });
```

Comment:
This code can be optimized:

```php
$this->roots = array_filter(array_map(function (string $root) use ($allowUnresolvable): ?string {
    return $this->sanitizeRootPath($root, $allowUnresolvable);
}, $roots));
```
All paths that can be converted to [FALSE](http://php.net/manual/en/function.array-filter.php) are not correct.